plugins/search/src/components/SearchContext/useInfiniteScrollSearch.ts

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright 2022 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { UIEventHandler, UIEvent, useCallback, useEffect } from 'react';
+import { useSearch } from './SearchContext';
+
+export function useInfiniteScrollSearch() {
+  const { fetchNextPage } = useSearch();
+  const onScroll = useCallback<UIEventHandler<HTMLElement>>(
+    e => {
+      const element = e.target as HTMLElement;
+
+      const isEndReached =
```

Comment:
sure, good point. I could pass a configurable fixed pixes (like an offset) as you proposed or a number between (0-100) referring to where in the page we should start loading more elements (ex when the user reach 75% of the scrollable content).
What do you think?

Reply:
the height of the Container is up to the user, together with the height of each rendered item. If we set let's say, 100px it might be a problem if the user will render a very tiny list on a small portion of the page

Reply:
Hmm - is the height of the container relevant? Shorter containers will just have more "screens" of scrollable area. If we're waiting until the user is (say) 500px from the bottom, they will won't hit that point until the user scrolls all the way down, even if the container is 400px tall. 

Having said that I do think you're right - it seems better to define the cutoff as a percentage of the overall scrollable area, that way we're not sensitive to differences in the height of the list. We should perhaps even factor in the number of results? We could do something like loading more when the user sees the 5th-to-last result, regardless of the number of results shown. That way we get a relatively consistent experience regardless of the page size (which can vary in different search engines). 

To be clear - I have no strong preference here, I'm broadly happy with all options proposed so far! Just a bit of friendly bikeshedding 😁.

Reply:
I think this is now irrelevant after what @iamEAP suggested below :)